orchestra/bots/tests/test_staffbot.py

```diff
@@ -31,36 +48,141 @@ def test_commands(self):
         response = bot.dispatch(mock_slack_data)
         self.assertTrue(bot.default_error_text in response.get('text', ''))
 
-    def test_staff_command(self):
+    def test_staff_command_errors(self):
+        """
+        Test that the staffing logic errors are raised during
+        staff command.
+        """
+        bot = StaffBot()
+        data = get_mock_slack_data(text='staff 999999999999')
+
+        response = bot.dispatch(data)
+        self.assertEqual(response.get('text'),
+                         bot.task_does_not_exist_error.format('999999999999'))
+
+        data = get_mock_slack_data(text='staff')
+        response = bot.dispatch(data)
+        self.assertTrue(bot.default_error_text in response.get('text'))
+
+        task = TaskFactory(status=Task.Status.COMPLETE)
+        data = get_mock_slack_data(text='staff {}'.format(task.id))
+        response = bot.dispatch(data)
+        self.assertEquals(response.get('text'),
+                          bot.task_assignment_error
+                          .format(task.id,
+                                  'Status incompatible with new assignment'))
+
+    @patch('orchestra.bots.staffbot.StaffBot._send_staffing_request_by_mail')
+    @patch('orchestra.bots.staffbot.StaffBot._send_staffing_request_by_slack')
+    def test_staff_command(self, mock_slack, mock_mail):
         """
         Test that the staffing logic is properly executed for the
         staff command.
         """
+        task = (Task.objects
+                .filter(status=Task.Status.AWAITING_PROCESSING)
+                .first())
+
+        # Get certified worker
+        worker = self._get_worker_for_task(
+            task, WorkerCertification.Role.ENTRY_LEVEL)
+
+        # Get worker communication preference
+        self._test_staffing_requests(worker, task, 'staff {}'.format(task.id),
+                                     False, False)
```

Comment:
i would pass these args by name e.g. ` self._test_staffing_requests(worker, task, 'staff {}'.format(task.id), can_mail=False, can_slack=False)` just so it's clear what the bools are